exercises/04-Build-Agents/README.md

```diff
@@ -1,8 +1,76 @@
+**Creating a Jenkins Agent**
 =====================================================
 
+Build agents are how you'll be running most of your Jenkins jobs. Not only is it considered a [best practice](https://www.jenkins.io/doc/book/using/best-practices/#build-on-agents), but it means you can also tightly control the build environment your jobs run on without installing anything on your Jenkins controller.
 
+In a production Jenkins instance, all of your jobs will run on agents insetad of the controller, but since this is a lab environment, you'll only be using them in this exercise for this course.
 
-**Step 1: Setup a Freestyle Project**
+In this exercise, you'll connect an agent to the Jenkins controller, label the agent, then run a job that uses that agent.
+
+**Step 1: Create the Node Agent**
+-------------------------------------------
+
+First, you need to tell the Jenkins controller to expect to use a Jenkins agent running on a Node. Nodes can be other physical computers, virtual machines, Kubernetes pods, or even docker containers, as long as they are on the same network as the controller. 
+
+You can setup a new node agent by navigating to **Manage Jenkins** > **Nodes** and clicking on **New Node**.
+
+Give it the name `docker-agent` and select the radio button to make it a **Permenant Agent**, then click **Create**
+
+On the next window, fill out the following parameters and leave the rest as they are:
+
+- **Remote root directory**: `/var/jenkins`
+- **Labels**: `agent`
+
+Click **Save, then from the nodes window, click on the new node. That should show you the commands to connect it to the controller via JNLP once you have another node running on the same network. Save the commands to connect a Unix agent and proceed to the next step.
+
+**Step 2: Connect the Agent to the Controller**
+-------------------------------------------
+
+Surprise, you've already got a Jenkins node running in your docker environment! 
+
+Agents are required to have network connectivity to the controller, so for a lab environment like this, the agent has to run in a docker container on the same network.
+
+> Note: Some of the commands in this exercise will change `localhost` to the name of a container in the same network. You ideally won't have to do that in a real Jenkins environment that has been configured correctly, but it's unfortunatley unavoidable in a lab such as this. Please let the instructor know if you need any help with this!
+
+Start by opening a shell on the agent using the command below:
+
+`docker exec -it agent /bin/bash`
+
+You can get the docker container ID by running `hostname` from this shell window. Save that for later, you'll need it in a later step.
+
+Pull up the commands from the end of step 1, and replace all instances of `localhost` with `jenkins`. This is because that is the container name of the jenkins controller defined in the docker compose file, and the agent will be able to find it on the docker network.
```

Comment:
it's unclear what commands you're talking about here, so it's hard to follow the rest of this guide